Ghost follow is now an orbit!

MrsTonedOne opened the pull request.

MrsTonedOne committed:
Ghost follow is now an orbit!

Steelpoint commented:
Let me see people easier, what's not to like :+1:

WJohn commented:
This is adorable.

MrsTonedOne committed:
Fixes bug where orbiting while orbiting would break orbits

daman997 commented:
do it :+1:

RemieRichards commented on code/__HELPERS/unsorted.dm:
```diff
@@ -1423,6 +1423,9 @@ B --><-- A
 		return
+	if (orbiting)
+		stop_orbit()
+		sleep 1 //sadly this is the only way to ensure the original orbit proc stops and resets the mob's transform.
```
spooky, didn't know it accepted sleep as a keyword instead of a proc.

Anonus commented:
Bonus; this means following someone won't have a delay as the trail of ghosts tries to catch up.
:+1:

MrsTonedOne committed:
Some code clean up

MrsTonedOne committed:
I slept so I should make sure A still exists

Dorsisdwarf commented:
Can we haz gifs?

MrPerson commented on code/__HELPERS/unsorted.dm:
```diff
@@ -1423,6 +1423,11 @@ B --><-- A
+		stop_orbit()
+		sleep(1) //sadly this is the only way to ensure the original orbit proc stops and resets the atom's transform.
+		if (orbiting || !istype(A)) //post sleep re-check
+			return 
```
You may want to make sure A hasn't been deleted instead of just doing an istype()

RemieRichards replied:
```istype()``` will fail on null, so if it's truly deleted it's fine.
```istype()``` obviously doesn't cover qdel-limbo, but the object still exists during that so it should probably be fine.

MrPerson replied:
I'll admit that being qdeleted within a decisecond is unlikely, but it could happen

Anonus replied:
If it can, it WILL.

MrStonedOne replied:
I fixed it in a much more practical way. (look down about 5 lines)

Iamgoofball commented:
rip omnighost
:+1:

MrsTonedOne committed:
Fixes orbiting a deleted atom sending you to null space

This also handles the garbage collection issues. As far as i'm concerned, if its been qdeled, but loc isn't null, it's fair game for orbits.
